app/src/Event/EventController.php

```diff
@@ -291,6 +302,85 @@ public function submit()
     }
 
     /**
+     * Action used to display a form to edit an event and with which the form can be submitted
+     *
+     * @return void
+     */
+    public function edit($friendly_name)
+    {
+        $request = $this->application->request();
+
+        $eventApi = $this->getEventApi();
+        $event    = $eventApi->getByFriendlyUrl($friendly_name);
+        if (! $event) {
+            $this->redirectToListPage();
+        }
+
+        if (! $this->userIsAllowedToEdit($event)) {
+            $this->redirectToDetailPage($event->getUrlFriendlyName());
+        }
+
+        /** @var FormFactoryInterface $factory */
+        $factory = $this->application->formFactory;
+        $form    = $factory->create(new EventFormType(), $event->setTags(implode(', ', $event->getTags())));
+        if ($request->isPost()) {
+            $form->submit($request->post($form->getName()));
+
+            if ($form->isValid()) {
+                $event = $this->editEventUsingForm($form, $event);
+
+                if ($event instanceof EventEntity) {
+                    $this->redirectToDetailPage($event->getUrlFriendlyName());
+                }
+
+                // held for moderation
+                if ($event === null) {
+                    $this->redirectToListPage();
+                }
+            }
+        }
+
+        $this->render(
+            'Event/edit.html.twig',
+            array(
+                'event'     => $event,
+                'form'      => $form->createView(),
+                'timezones' => EventFormType::getNestedListOfTimezones(),
+            )
+        );
+
+
+    }
+
+    /**
+     * @param array $event
+     *
+     * @return bool
+     */
+    protected function userIsAllowedToEdit($event)
+    {
+        if (! isset($_SESSION['user'])) {
+            return false;
+        }
+
+        if (isset($event['can_edit'])) {
```

Comment:
$event is an instance of EventEntity here, so we can't access it like an array. You'll need a `getCanEdit()` method on the EventEntity.